vault/expiration.go

```diff
@@ -147,15 +157,19 @@ func expireLeaseStrategyRevoke(ctx context.Context, m *ExpirationManager, le *le
 // using a given view, and uses the provided router for revocation.
 func NewExpirationManager(c *Core, view *BarrierView, e ExpireLeaseStrategy, logger log.Logger) *ExpirationManager {
 	exp := &ExpirationManager{
-		core:       c,
-		router:     c.router,
-		idView:     view.SubView(leaseViewPrefix),
-		tokenView:  view.SubView(tokenViewPrefix),
-		tokenStore: c.tokenStore,
-		logger:     logger,
-		pending:    sync.Map{},
-		leaseCount: 0,
-		tidyLock:   new(int32),
+		core:        c,
+		router:      c.router,
+		idView:      view.SubView(leaseViewPrefix),
+		tokenView:   view.SubView(tokenViewPrefix),
+		tokenStore:  c.tokenStore,
+		logger:      logger,
+		pending:     sync.Map{},
+		nonexpiring: sync.Map{},
+		leaseCount:  0,
+		tidyLock:    new(int32),
+
+		uniquePolicies:      make(map[string][]string),
+		emptyUniquePolicies: time.NewTicker(7 * 24 * time.Hour),
```

Comment:
Yeah... I thought about it.  It seems like it would be substantial additional overhead, because go-cache does its own locking, and tracks expiration per entry. We really only need to expire because we don't want indefinite growth.

For the number of distinct policies we have, maybe the per-entry costs are fine, but it seemed to introduce a lot of uncertainty.